gcovr/formats/cobertura/read.py

```diff
@@ -0,0 +1,141 @@
+    get_merge_mode_from_options,
+    insert_branch_coverage,
+    insert_file_coverage,
+    insert_line_coverage,
+)
+
+LOGGER = logging.getLogger("gcovr")
+
+
+#
+#  Get coverage from already existing gcovr JSON files
+#
+def read_report(options: Options) -> CovData:
+    """merge a coverage from multiple reports in the format
+    compatible with Cobertura"""
+
+    if len(options.cobertura_add_tracefile) == 0:
+        return None
+
+    datafiles = set()
+
+    for trace_files_regex in options.cobertura_add_tracefile:
+        trace_files = glob(trace_files_regex, recursive=True)
+        if not trace_files:
+            raise RuntimeError(
+                "Bad --add-tracefile option.\n" "\tThe specified file does not exist."
+            )
+        else:
+            for trace_file in trace_files:
+                datafiles.add(os.path.normpath(trace_file))
+
+    covdata = dict()
+    for filename in datafiles:
+        LOGGER.debug(f"Processing XML file: {filename}")
+
+        try:
+            root = etree.parse(filename).getroot()
+        except Exception:
+            raise RuntimeError(
+                "Bad --add-tracefile option.\n"
+                f"\tThe file {filename} is not a valid XML file."
+            )
+
+        for gcovr_file in root.xpath("./packages//class"):
+            if gcovr_file.get("filename") is None:
+                LOGGER.warning(
+                    f"missing filename attribute in class element of {filename}"
+                )
+                continue
+
+            file_path = os.path.join(
+                os.path.abspath(options.root),
+                os.path.normpath(gcovr_file.get("filename")),
+            )
+
+            filtered, excluded = apply_filter_include_exclude(
+                file_path, options.filter, options.exclude
+            )
+
+            # Ignore if the filename does not match the filter
+            if filtered:
+                LOGGER.debug(f"  Filtering coverage data for file {file_path}")
+                continue
+
+            # Ignore if the filename matches the exclude pattern
+            if excluded:
+                LOGGER.debug(f"  Excluding coverage data for file {file_path}")
+                continue
+
+            file_coverage = FileCoverage(file_path)
+            merge_options = get_merge_mode_from_options(options)
+            for xml_line in gcovr_file.xpath("./lines//line"):
+                insert_line_coverage(file_coverage, _line_from_xml(filename, xml_line))
+
+            insert_file_coverage(covdata, file_coverage, merge_options)
+
+    return covdata
+
+
+def _line_from_xml(filename: str, xml_line) -> LineCoverage:
+    try:
+        lineno = int(xml_line.get("number"))
+        count = int(xml_line.get("hits"))
+        is_branch = xml_line.get("branch") == "true"
+        branch_msg = xml_line.get("condition-coverage")
+    except Exception:
+        LOGGER.warning(f"Invalid line information in file {filename}")
```

Comment:
In this case I think we shouldn't catch the exception since we can't continue.

Reply:
I think the raw exception message could be a bit vague. So I'll update it to rethrow with a nicer message, and provide the xml line that caused the issue.

Reply:
Or add the raw message to the existing one.

Reply:
I mean this:
```suggestion
        raise RuntimeError(
            "Bad --covertura-add-tracefile option.\n"
            f"\tCorrupt/missing line coverage: {etree.tostring(xml_line).decode()}\n"
            f"\t{e}"
        )
```
So give a nicer looking message, with the line that caused it, but include the raw message at the end. How does that look?

Reply:
You're right, here can several errors occur. E.g. the attribute doesn't exist or has a different type. Would it be better to read each attribute in a separate try block?

Reply:
On closer inspection, only the 'int' conversion will throw here. A missing attribute will return None. I'll check those two conversions separately and give a specific message